targets/csharp-unityv2/source/Shared/Internal/PlayFabHttp/PlayFabHTTP.cs

```diff
@@ -0,0 +1,337 @@
+            _internalHttp.Awake();
+        }
+
+        /// <summary>
+        /// Internal method for Make API Calls
+        /// </summary>
+        /// <typeparam name="TRequestType"></typeparam>
+        /// <typeparam name="TResultType"></typeparam>
+        /// <param name="api"></param>
+        /// <param name="apiEndpoint"></param>
+        /// <param name="request"></param>
+        /// <param name="authType"></param>
+        /// <param name="resultCallback"></param>
+        /// <param name="errorCallback"></param>
+        /// <param name="customData"></param>
+        protected internal static void MakeApiCall<TRequestType, TResultType>(string api, string apiEndpoint,
+            TRequestType request, string authType, Action<TResultType> resultCallback,
+            Action<PlayFabError> errorCallback, object customData = null)
+        {
+            InitializeHttp();
+            SendEvent(request, null, ApiProcessingEventType.Pre);
+            _internalHttp.MakeApiCall(api, apiEndpoint, request, authType, resultCallback, errorCallback, customData);
+        }
+
+        /// <summary>
+        /// Monobehaviour Awake Method, gathers system data
+        /// </summary>
+        public override void Awake()
+        {
+            _gatherer.GatherData();
+            base.Awake();
+        }
+
+        /// <summary>
+        /// Monobehaviour OnEnable Method, registers logger
+        /// </summary>
+        public void OnEnable()
+        {
+            StartCoroutine(RegisterLogger());
+        }
+
+        /// <summary>
+        /// Monobehaviour OnDisable method, performs cleanup of Logger and bindings to Log Messages
+        /// </summary>
+        public void OnDisable()
+        {
+            if (!string.IsNullOrEmpty(PlayFabSettings.LoggerHost))
+            {
+                _logger = null;
+#if UNITY_5
+                UnityEngine.Application.logMessageReceivedThreaded -= HandleLogOutput;
+#else
+                UniytEngine.Application.RegisterLogCallback(null);
+#endif
+            }
+        }
+
+        /// <summary>
+        /// Monobehaviour Update method, logs System info data if gathered.
+        /// </summary>
+        public void Update()
+        {
+            _internalHttp.Update();
+            if (!_systemInfoLogged && PlayFabSettings.EnableRealTimeLogging)
+            {
+                _gatherer.EnqueueToLogger(_logMessageQueue);
+                _systemInfoLogged = true;
+            }
+        }
+
+        /// <summary>
+        /// Monobehaviour FixedUpdate, logs messages when enabled.
+        /// </summary>
+        void FixedUpdate()
+        {
+
+            lock (_logMessageLock)
+            {
+                if (_logMessageQueue.Count > 0 && PlayFabSettings.EnableRealTimeLogging)
+                {
+                    if (_logger == null)
+                    {
+                        return;
+                    }
+                    _logger.Open();
```

Comment:
This looks like you're writing to the log system from the unity thread... which is seems the opposite of your intent.  I think you want to do this work in the other thread right?